File: be_camphub/src/main/java/org/camphub/be_camphub/dto/response/booking/BookingResponse.java
package org.camphub.be_camphub.dto.response.booking;

import java.util.UUID;

import lombok.*;
import lombok.experimental.FieldDefaults;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class BookingResponse {
    UUID id;
    UUID lesseeId;
    UUID lessorId;
    UUID itemId;

    String itemName;
    String lessorName; // tên đầy đủ của người cho thuê
    String lesseeName; // tên đầy đủ của người thuê

    Integer quantity;
    Double pricePerDay;
    Double depositAmount;
    Double totalAmount;
    String startDate;
    String endDate;
    String note;
    String status;
    String createdAt;
    Boolean hasReviewed;
}

File: fe_camphub/src/libs/core/dto/response/booking.response.ts
import { BookingStatus } from "../../constants";


export interface BookingResponse {
    id: string;
    lesseeId: string;
    lessorId: string;
    itemId: string;

    itemName: string;
    lessorName: string;
    lesseeName: string;

    quantity: number;
    pricePerDay: number;
    depositAmount: number;
    totalAmount: number;

    startDate: string;       // yyyy-MM-dd
    endDate: string;         // yyyy-MM-dd
    
    note: string;
    status: BookingStatus;
    createdAt: string;       // ISO datetime
    hasReviewed: boolean;
}
